src/app/modules/shared/utils/project-storage.util.ts

```diff
@@ -0,0 +1,30 @@
+import { Project } from '../models/project.model';
+import { PROJECTS_KEY_FOR_LOCAL_STORAGE } from '../../../../environments/environment';
+
+const projectsKey = PROJECTS_KEY_FOR_LOCAL_STORAGE;
+
+export function updateProjectStorage(serverProjects: Project[]): void {
+  const storageProjects: Project[] = getProjectsOnStorage(projectsKey);
+  const isServerProjectsNotEmpty = serverProjects && serverProjects.length !== 0;
+  const updatedStorageProjects: Project[] = [];
+
+  if (serverProjects && isServerProjectsNotEmpty && storageProjects) {
+    storageProjects.forEach((storageProject: Project) => {
+      const project = serverProjects.find((serverProject) => serverProject.id === storageProject.id);
+
+      if (project) {
+        updatedStorageProjects.push(project);
+      }
+    });
+
+    const projectsForLocalStorage = JSON.stringify(updatedStorageProjects);
+    localStorage.setItem(projectsKey, projectsForLocalStorage);
+  }
+}
+
+export function getProjectsOnStorage(projectsIdentifier: string): Project[] {
+  const projectsInsideLocalStorage: string = localStorage.getItem(projectsIdentifier);
+  return projectsInsideLocalStorage && JSON.parse(projectsInsideLocalStorage);
```

Comment:
Here you can use lodash to get the value, I think is more clear when you use the next `getItem(object, 'path', defaultValue)` and you can prevent errors

Reply:
Same comment as above. There are two possibilities described when using `localStorage.getItem()`, you can return the value as such or just null, see here. https://developer.mozilla.org/en-US/docs/Web/API/Storage/getItem, so I think it is unnecessary to use lodash for this case.